scripts/BindingGenerator.lua

```diff
@@ -0,0 +1,446 @@
+local arbProcedures = procedures:Where( function(x) return x.vendor == "ARB"; end );
+
+-- Only consider core and ARB functions.
+local nameList = coreProcedures:Join( arbProcedures ):Select(
+	function(p)
+		return p.name;
+	end );
+
+local nameSet = {};
+local uniqueNames = List();
+
+for s, k in ipairs( nameList ) do
+	if not nameSet[k] then
+		nameSet[k] = true;
+		uniqueNames:Add( k );
+	end
+end
+
+for i=1, #procedures do
+	local p = procedures[i];
+	procTable[p.name] = procTable[p.name] or {};
+	local key = p.vendor or "core";
+	procTable[p.name][key] = p;
+end
+
+local priorityList = List{ "core", "ARB", "OES", "KHR" };
+
+local typedefs = List();
+local pointers = List();
+local loader = List();
+
+for s, str in ipairs( uniqueNames ) do
+	pointers:Add( ("\t%s %s = NULL;")( pfnFormat( str:upper() ), str ) );
+	local typeDefGenerated = false;
+	for i=1, #priorityList do
+		local k = priorityList[i];
+		local proc = procTable[str][k]
+		if proc then
+			if not typeDefGenerated then
+				typedefs:Add( GetProcedureTypedef( proc ) );
+				typeDefGenerated = true;
+			end
+			local vendor = k == "core" and "" or k;
+			loader:AddFormat(
+				'\tif (!%s) %s = (%s)cmgr->getProcAddress("%s");\n',
+				str, str, pfnFormat( proc.name:upper() ), ("gl%s%s")(str,vendor)
+			);
+		end
+	end
+end
+
+
+------------ Write files ------------
+
+-- Write loader header
+local f = io.open( sourceTreePath .. "/include/mt_opengl.h", "wb" );
+f:write[[
+// This code was generated by scripts/BindingGenerator.lua
+// Do not modify it, modify and run the generator instead.
+
+#pragma once
+
+#include <string>
+#include <unordered_set>
+#include "irrTypes.h"
+#include "IContextManager.h"
+#include <KHR/khrplatform.h>
+
+#ifndef APIENTRY
+	#define APIENTRY
+#endif
+#ifndef APIENTRYP
+	#define APIENTRYP APIENTRY *
+#endif
+#ifndef GLAPI
+	#define GLAPI extern
+#endif
+
+]];
+
+f:write[[
+class OpenGLProcedures {
+private:
+]];
+f:write( definitions:Concat( "\n" ) );
+f:write( "\n" );
+f:write[[
+	// The script will miss this particular typedef thinking it's a PFN,
+	// so we have to paste it in manually. It's the only such type in OpenGL.
+	typedef void (APIENTRY *GLDEBUGPROC)
+		(GLenum source,GLenum type,GLuint id,GLenum severity,GLsizei length,const GLchar *message,const void *userParam);
+
+]]
+f:write( typedefs:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write [[
+	std::unordered_set<std::string> extensions;
+	void* GetProcAddress();
+public:
+	// Call this once after creating the context.
+	void LoadAllProcedures(irr::video::IContextManager *cmgr);
+	// Check if an extension is supported.
+	inline bool IsExtensionPresent(const std::string &ext)
+	{
+		return extensions.find(ext) != extensions.end();
+	}
+
+]];
+f:write( pointers:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write( cppConsts:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write( "};\n" );
+f:write( "\n//Global GL procedures object.\n" );
+f:write( [[
+#ifdef WIN32
```

Comment:
(used everywhere else too)
```suggestion
#ifdef _WIN32
```